src/cfnlint/rules/mappings/ApproachingLimitName.py

```diff
@@ -0,0 +1,32 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+from cfnlint.helpers import LIMITS
+
+
+class LimitName(CloudFormationLintRule):
+    """Check maximum Mapping name size limit"""
+    id = 'I7011'
+    shortdesc = 'Mapping name limit'
+    description = 'Check the size of Mapping names in the template is approaching the upper limit'
+    source_url = 'https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html'
+    tags = ['mappings', 'limits']
+
+    def match(self, cfn):
+        """Check CloudFormation Mappings"""
+
+        matches = []
+
+        mappings = cfn.template.get('Mappings', {})
+
+        for mapping_name in mappings:
+            path = ['Mappings', mapping_name]
+            if len(mapping_name) > 0.9 * LIMITS['mappings']['name'] and not len(mapping_name) > LIMITS['mappings']['name']:
```

Comment:
I wonder if this would be more easy to read.
`if 0.9 * LIMITS['mappings']['name'] < len(mapping_name) <=  LIMITS['mappings']['name']:`

Reply:
@kddejong agreed, done: https://github.com/aws-cloudformation/cfn-python-lint/pull/1271/commits/dacbc2dcf144b0e40676b6bb8fe581d08c2ade98

also abstracted `LIMITS['threshold']` out to be more easily configurable in the future